docs/history-service.md

```diff
@@ -0,0 +1,274 @@
+
+- [`executions` table schema in Cassandra](https://github.com/temporalio/temporal/blob/ef49189005b5323c532264287af6c08a447aab8a/schema/cassandra/temporal/schema.cql#L7)
+- [handling](https://github.com/temporalio/temporal/blob/ef49189005b5323c532264287af6c08a447aab8a/service/history/timer_queue_active_task_executor.go#L114) of the various types of timers in the Timer Task queue.
+
+   </details>
+   <br>
+
+#### Transfer Task queue
+
+A Transfer Task is available to be executed immediately.
+When a Transfer Task is executed, an RPC is made to the Matching Service to enqueue the Workflow Task or Activity Task specified by the Transfer Task.
+
+<details>
+<summary><i>Code entrypoints</i></summary>
+
+- The [`Execute`](https://github.com/temporalio/temporal/blob/ef49189005b5323c532264287af6c08a447aab8a/service/history/transfer_queue_active_task_executor.go#L114) method on a task dispatches according to task type.
+
+- For an Activity, [`processActivityTask`](https://github.com/temporalio/temporal/blob/ef49189005b5323c532264287af6c08a447aab8a/service/history/transfer_queue_active_task_executor.go#L169) is called, resulting in [an RPC to Matching service](https://github.com/temporalio/temporal/blob/ef49189005b5323c532264287af6c08a447aab8a/service/history/transfer_queue_task_executor_base.go#L121) to create the Activity Task.
+
+- Similarly, for a workflow task: [`processWorkflowTask`](https://github.com/temporalio/temporal/blob/ef49189005b5323c532264287af6c08a447aab8a/service/history/transfer_queue_active_task_executor.go#L217) results in an [RPC to Matching Service](https://github.com/temporalio/temporal/blob/ef49189005b5323c532264287af6c08a447aab8a/service/history/transfer_queue_task_executor_base.go#L152) to create the Workflow Task.
+</details><br>
+
+#### Timer Task queue
+
+There are several scenarios in which it's necessary to set a timer associated with a workflow execution and do something when the timer expires. For example:
+
+- A Timer Task representing a `workflow.sleep()` call in user workflow code is processed, when the timer expires, by creating a Transfer Task specifying that a Workflow Task should be enqueued in Matching to advance the workflow.
+- A Timer Task is created on receiving a `ScheduleActivity` command, representing the [schedule-to-start](https://docs.temporal.io/activities#schedule-to-start-timeout) timeout.
```

Comment:
All timeouts have corresponding timer task.

Reply:
Right, it does say "For example:", and I wanted to give a concrete example here.